modules/pensions/lib/pensions/benefits_intake/pension_benefit_intake_job.rb

```diff
@@ -37,6 +39,18 @@ class PensionBenefitIntakeError < StandardError; end
       rescue
         claim = nil
       end
+
+      if claim.present? && Flipper.enabled?(:pension_event_bus_submission_enabled)
+        user_icn = UserAccount.find_by(id: claim&.user_account_id)&.icn.to_s
+
+        Kafka.submit_event(
```

Comment:
Perhaps you can refactor and call `submit_traceability_to_event_bus` with a state argument?

Reply:
@madebydna The name of the method can be whatever, I was trying to keep it a bit short but if you foresee other event submissions I can update it! 

As for the argument(s), it needs all 4 because of how we get the data (minus `submission_name` I suppose) between the actual job and the Sidekiq exhaustion. We don't have access to instance variables in the exhaustion block :/

Reply:
Okay, yes, if you no longer have access to the instance variables that it would not work.